### DISCUSSION TITLE
Virtualtext not showing as comments after colorscheme reload

### DISCUSSION THREAD

**Discussion by esan0** [w0rp/ale#4621] [Q&A - Ask for help with problems] [answered]
Hi folks,

## Issue
When the colorscheme gets reloaded, virtualtext loses its "comment" highlighting.

Clean vim loaded:
![virtualtext_1](https://github.com/dense-analysis/ale/assets/110071118/ae66105d-6449-41ed-81b6-20c0f0c44527)

reload colorscheme
![virtualtext_2](https://github.com/dense-analysis/ale/assets/110071118/d7a89149-2d7c-482f-9170-107edab47669)

I was able to get around this partially by adding the following to the function that manually toggles `background` (and thus reloads the colorscheme):
```vim
function! s:TogBG()
    let &background = ( &background == "dark"? "light" : "dark" )
    highlight link ALEVirtualTextError Comment
    highlight link ALEVirtualTextStyleError ALEVirtualTextError
    highlight link ALEVirtualTextWarning Comment
    highlight link ALEVirtualTextStyleWarning ALEVirtualTextWarning
    highlight link ALEVirtualTextInfo ALEVirtualTextWarning
    hi ALEError cterm=undercurl
    hi ALEWarning cterm=undercurl
endfunction

```

However, this *doesn't* solve the problem when the colorscheme is reloaded automatically, for example, when my terminal background gets changed and vim follows that change.

I'm also suspicious that if these highlight groups get somehow reset, then other non-visible items may also silently reset.
In fact, I also have to add ALEError and ALEWarning `cterm=undercurl` in the manual function, too, because they get reset to underline.

## Question
What is causing these highlight groups to be cleared when the colorscheme is reloaded?

## Environment
$ uname -r
6.5.5-arch1-1

$ vim --version
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Oct 01 2023 12:12:50)
Included patches: 1-1968
Compiled by Arch Linux

On `ale` commit `53b01d6`

**Comment by w0rp** [comment, marked as answer]
Create an issue for this.

**Reply by esan0** [reply]
Thanks!

Issue is here: https://github.com/dense-analysis/ale/issues/4622
